### PR TITLE
[PATCH v1] tests: compile and execute validation tests before the rest of tests

### PR DESCRIPTION
**Pull request by lumag**
Validation tests examine API thoroughly. Compile and execute them before
performance and misc tests.

Signed-off-by: Dmitry Eremin-Solenikov <dmitry.ereminsolenikov@linaro.org>

### CODE REVIEW

File: test/Makefile.am [hunk 1]
```diff
@@ -1 +1 @@
-SUBDIRS = common performance miscellaneous validation
+SUBDIRS = common validation performance miscellaneous
```
Comment (muvarov) [inline test/Makefile.am]: For parallel make case you need ".". Might be for the same time it's good to isolate performance tests. I.e.:
SUBDIRS = common validation . performance . miscellaneous
https://www.gnu.org/software/automake/manual/html_node/Subdirectories.html

Comment (lumag) [reply test/Makefile.am]: @muvarov hmm, you do not need "." here, as there is nothing executed in the current (test/) directory.

Comment (muvarov) [reply test/Makefile.am]: @lumag --enable-test-perf enables performance tests. On parallel with 'make check' it's better to not run them with validation tests at the same time.

Comment (lumag) [reply test/Makefile.am]: Dot just allows current directory to be built in parallel with the subdirs.

Comment (muvarov) [reply test/Makefile.am]: than it has to be ok.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**